source/hpack.c

```diff
@@ -246,6 +288,19 @@ struct aws_hpack_context *aws_hpack_context_new(struct aws_allocator *allocator,
         return NULL;
     }
 
+    if (aws_hash_table_init(
+            &context->dynamic_table.reverse_lookup_name_only,
+            allocator,
+            max_dynamic_elements,
+            aws_hash_byte_cursor_ptr,
+            (aws_hash_callback_eq_fn *)aws_byte_cursor_eq,
+            NULL,
+            NULL)) {
+        aws_hash_table_clean_up(&context->dynamic_table.reverse_lookup);
+        aws_mem_release(allocator, context->dynamic_table.buffer);
+        return NULL;
```

Comment:
your leaking the `context`. The if-statement above this one leaks it too
I suggest the usual `goto error` garbage